src/Simple.OData.Client.Core/Fluent/FluentCommandDetails.cs

```diff
@@ -47,6 +48,7 @@ public FluentCommandDetails(FluentCommandDetails parent, ConcurrentDictionary<ob
             this.OrderbyColumns = new List<KeyValuePair<string, bool>>();
             this.MediaProperties = new List<string>();
             this.BatchEntries = batchEntries;
+            this.Headers = new Dictionary<string, string>();
```

Comment:
You lazily create this object above, but also set it here. I think this assignment is unnecessary, and if removed would make this feature "pay for play", rather than adding another allocation for requests even lacking headers.

Reply:
I decided to respect the pattern in FluentCommandDetails class and assign the dictionary in the constructor as the other collections in the class. Fixed the missing assignment in the cloning constructor below. Removed the extra allocations in WithHeaders implementation.